client/network/src/request_responses.rs

```diff
@@ -0,0 +1,866 @@
+		RequestResponseMessage, ResponseChannel, ProtocolSupport
+	},
+	swarm::{
+		protocols_handler::multi::MultiHandler, NetworkBehaviour, NetworkBehaviourAction,
+		PollParameters, ProtocolsHandler,
+	},
+};
+use std::{
+	borrow::Cow, collections::{hash_map::Entry, HashMap}, io, iter, pin::Pin,
+	task::{Context, Poll}, time::Duration,
+};
+
+pub use libp2p::request_response::{InboundFailure, OutboundFailure, RequestId};
+
+/// Configuration for a single request-response protocol.
+#[derive(Debug, Clone)]
+pub struct ProtocolConfig {
+    /// Name of the protocol on the wire. Should be something like `/foo/bar`.
+    pub name: Cow<'static, str>,
+
+    /// Maximum allowed size, in bytes, of a request.
+    ///
+    /// Any request larger than this value will be declined as a way to avoid allocating too
+    /// much memory for the it.
+    pub max_request_size: usize,
+
+    /// Maximum allowed size, in bytes, of a response.
+    ///
+    /// Any response larger than this value will be declined as a way to avoid allocating too
+    /// much memory for the it.
+    pub max_response_size: usize,
+
+    /// Duration after which emitted requests are considered timed out.
+    ///
+    /// If you expect the response to come back quickly, you should set this to a smaller duration.
+    pub request_timeout: Duration,
+
+    /// Channel on which the networking service will send incoming requests.
+    ///
+    /// Every time a peer sends a request to the local node using this protocol, the networking
+    /// service will push an element on this channel. The receiving side of this channel then has
+    /// to pull this element, process the request, and send back the response to send back to the
+    /// peer.
+    ///
+    /// The size of the channel has to be carefully chosen. If the channel is full, the networking
+    /// service will discard the incoming request send back an error to the peer. Consequently,
+    /// the channel being full is an indicator that the node is overloaded.
+    ///
+    /// You can typically set the size of the channel to `T / d`, where `T` is the
+    /// `request_timeout` and `d` is the expected average duration of CPU and I/O it takes to
+    /// build a response.
+    ///
+    /// Can be `None` if the local node does not support answering incoming requests.
+    /// If this is `None`, then the local node will not advertise support for this protocol towards
+    /// other peers. If this is `Some` but the channel is closed, then the local node will
+    /// advertise support for this protocol, but any incoming request will lead to an error being
+    /// sent back.
+    pub requests_processing: Option<mpsc::Sender<IncomingRequest>>,
+}
+
+/// A single request received by a peer on a request-response protocol.
+#[derive(Debug)]
+pub struct IncomingRequest {
+	/// Who sent the request.
+	pub origin: PeerId,
+
+	/// Request sent by the remote. Will always be smaller than
+	/// [`RequestResponseConfig::max_response_size`].
```

Comment:
```suggestion
	/// [`RequestResponseConfig::max_request_size`].
```

Or am I missing something? I can't find the comparison to the `max_response_size`.